xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/UserMentionEventListener.java

```diff
@@ -0,0 +1,200 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.mentions.MentionLocation;
+import org.xwiki.mentions.MentionsConfiguration;
+import org.xwiki.mentions.events.MentionEvent;
+import org.xwiki.mentions.events.MentionEventParams;
+import org.xwiki.mentions.events.NewMentionsEvent;
+import org.xwiki.mentions.notifications.MentionNotificationParameter;
+import org.xwiki.mentions.notifications.MentionNotificationParameters;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.observation.EventListener;
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.observation.event.Event;
+import org.xwiki.rendering.block.XDOM;
+
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.DocumentRevisionProvider;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseStringProperty;
+import com.xpn.xwiki.objects.PropertyInterface;
+
+import static java.util.Collections.singleton;
+import static java.util.Collections.singletonList;
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * Event listener for the mentions to local wiki users.
+ *
+ * @version $Id$
+ * @since 12.10RC1
+ */
+@Component
+@Named(UserMentionEventListener.TYPE)
+@Singleton
+public class UserMentionEventListener implements EventListener
+{
+    /**
+     * Type of the component.
+     */
+    public static final String TYPE = "UserMentionEventListener";
+
+    @Inject
+    private ObservationManager observationManager;
+
+    @Inject
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Inject
+    private QuoteService quote;
+
+    @Inject
+    private MentionsConfiguration configuration;
+
+    @Inject
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    @Inject
+    private DocumentRevisionProvider documentRevisionProvider;
+
+    @Inject
+    private MentionXDOMService xdomService;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public String getName()
+    {
+        return TYPE;
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        return singletonList(new NewMentionsEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        // We send notification to the mentions explicitly typed as local user, but also untyped mentions (i.e., with a
+        // null type).
```

Comment:
So I guess this answers my comment before about the handle of old mentions. Can you confirm?

Reply:
You are right, `MentionedActorReference` is initialized with a `null` type if the value is not defined in the macro. Mentions with a `null` type are then handled here.